src/App.tsx

```diff
@@ -1,14 +1,47 @@
-/* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
+import React, {
+  useState,
+  useMemo,
+  useEffect,
+  useContext,
+} from 'react';
 import '@fortawesome/fontawesome-free/css/all.css';
+import 'bulma/css/bulma.css';
 
-import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
-import { Loader } from './components/Loader';
+import {
+  TodoList,
+  TodoFilter,
+  TodoModal,
+  Loader,
+} from './components';
+import { getFilteredTodos } from './helpers/getFilteredTodos';
+import { Todo, Filters } from './types';
+import { getTodos } from './api';
+import { TodoContext } from './TodoContext';
 
 export const App: React.FC = () => {
+  const [todoItems, setTodoItems] = useState<Todo[]>([]);
+  const [filterByCategory, setFilterByCategory] = useState(Filters.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState('');
+
+  const visibleTodos = useMemo(() => getFilteredTodos(
+    filterByCategory,
+    query,
+    todoItems,
+  ), [query, filterByCategory, todoItems]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodoItems)
+      .catch((error) => {
+        throw error;
```

Comment:
In `catch` better to print some message to console or to show some error message on UI